numba/targets/builtins.py

```diff
@@ -30,8 +30,32 @@ def generic_is(context, builder, sig, args):
     """
     Default implementation for `x is y`
     """
-    return cgutils.false_bit
-
+    lhs_type, rhs_type = sig.args
+    lhs, rhs = args
+    # the lhs and rhs have the same type
+    if lhs_type == rhs_type:
+        lhs_mis = context.get_nrt_meminfo_recur(builder, lhs_type, lhs)
+        rhs_mis = context.get_nrt_meminfo_recur(builder, rhs_type, rhs)
+        if context.enable_nrt and lhs_mis and rhs_mis:
+            # check that the meminfo pointer matches
```

Comment:
Do you think that's useful? NRT-managed objects (e.g. lists) should have their own `is` implementation, especially as other details may differ (for example, two array views can point to the same NRT-allocated data but have different shapes).